src/main/java/net/sf/marineapi/nmea/io/AbstractDataReader.java

```diff
@@ -62,6 +62,10 @@ public boolean isRunning() {
 		return isRunning;
 	}
 
+	protected void setRunning(boolean running) {
```

Comment:
There's already `stop()` method for doing this. Also, notice that except for stopping, you can't control running from outside. That is, `setRunning(true)` won't start reader again if the underlying stream is already dead.